admin/backend/knowledge_refresh_service.py

```
@@ -0,0 +1,166 @@
+"""
+Knowledge base refresh service for production-ready re-indexing.
+"""
+
+import asyncio
+import logging
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, Optional, Tuple
```

Ruff (0.12.2)

`typing.Tuple` imported but unused

Remove unused import: `typing.Tuple`

(F401)